src/components/sections/ProductShelf/ProductShelf.tsx

```diff
@@ -1,34 +1,40 @@
 import React from 'react'
-import type { ProductSummary_ProductFragment } from '@generated/graphql'
 import ProductShelfSkeleton from 'src/components/skeletons/ProductShelfSkeleton'
+import { useProductsQuery } from 'src/sdk/product/useProductsQuery'
+import type { ProductsQueryQueryVariables } from '@generated/graphql'
 
 import ProductCard from '../../product/ProductCard'
 import Section from '../Section'
 
 import './product-shelf.scss'
 
-interface ProductShelfProps {
-  products: ProductSummary_ProductFragment[]
+interface ProductShelfProps extends ProductsQueryQueryVariables {
   title: string | JSX.Element
   withDivisor?: boolean
 }
 
 function ProductShelf({
-  products,
   title,
   withDivisor = false,
+  ...variables
 }: ProductShelfProps) {
+  const products = useProductsQuery(variables)
+
+  if (products?.edges.length === 0) {
+    return null
+  }
+
   return (
     <Section
       className={`layout__section ${withDivisor ? 'shelf__divisor' : ''}`}
     >
       <h2 className="text__title-section layout__content">{title}</h2>
       <div data-product-shelf-content>
-        <ProductShelfSkeleton loading={products.length === 0}>
+        <ProductShelfSkeleton loading={products == null}>
           <ul data-product-shelf-items className="layout__content">
-            {products.map((product, idx) => (
-              <li key={`${product.id}`}>
-                <ProductCard product={product} index={idx + 1} />
+            {products?.edges.map((product, idx) => (
```

Comment:
```suggestion
            {products.edges.map((product, idx) => (
```
Not needed, since it's already verified. If the ts complain, need to fix the first conditional at line 23.

Reply:
Did you fixed the line 23?

Reply:
what do you think we should use for the conditional of line 23? We can only return `null` when there is no products, a.k.a, products.length === 0. When it's loading, we should render the component for the skeleton take place

Reply:
Ok, np, but this is weird form, is it for you?

Reply:
Oh i see...
products?.edge.length can be undefined, right? You need to check like this
```tsx
!products || products.edge.length === 0
```

Reply:
yes, but that's the thing. I don't want it to return null when products is undefined. When it's undefined it needs to render the skeleton